databuilder/query_engines/spark.py

```diff
@@ -59,6 +59,20 @@ def temp_table_needs_dropping(self, create_table_query):
     def get_temp_database(self):
         return self.backend.temporary_database
 
+    def _convert_date_diff_to_days(self, start, end):
+        """
+        Calculate difference in days
+        The sparkSQL datediff function only calculate diff in days, and take the
+        parameters in the order (end, start)
+        """
+        return sqlalchemy.func.datediff(end, start)
+
+    def _convert_date_diff_to_weeks(self, start, end):
```

Comment:
This is the same as `BaseSQLQueryEngine._convert_date_diff_to_weeks`, is it not? Do we duplicate it here to ensure `SparkQueryEngine._convert_date_diff_to_days` is called in preference to `BaseSQLQueryEngine._convert_date_diff_to_days`? (i.e. to ensure `self` is an instance of `SparkQueryEngine` rather than `BaseSQLQueryEngine`?)

Reply:
What Iain said. It would be a shame to have to leave this duplication in.

Reply:
No, there's no reason for the duplication. In fact, I've already taken it out in the the next thing I'm working on locally, but I can add that to this PR